examples/rooch_blog/sources/article.move

```diff
@@ -0,0 +1,223 @@
+module rooch_blog::article {
+    use moveos_std::event;
+    use moveos_std::object::{Self, Object};
+    use moveos_std::object_id::ObjectID;
+    use moveos_std::object_storage;
+    use moveos_std::storage_context::{Self, StorageContext};
+    use moveos_std::tx_context;
+    use std::error;
+    use std::option;
+    use std::signer;
+    use std::string::String;
+    friend rooch_blog::rooch_blog;
+
+    const EID_DATA_TOO_LONG: u64 = 102;
+    const EINAPPROPRIATE_VERSION: u64 = 103;
+    const ENOT_GENESIS_ACCOUNT: u64 = 105;
+
+    public fun initialize(storage_ctx: &mut StorageContext, account: &signer) {
+        assert!(signer::address_of(account) == @rooch_blog, error::invalid_argument(ENOT_GENESIS_ACCOUNT));
+        let _ = storage_ctx;
+        let _ = account;
+    }
+
+    struct Article has key {
+        version: u64,
+        title: String,
+        body: String,
+    }
+
+    /// get object id
+    public fun id(article_obj: &Object<Article>): ObjectID {
+        object::id(article_obj)
+    }
+
+    public fun version(article_obj: &Object<Article>): u64 {
+        object::borrow(article_obj).version
+    }
+
+    public fun title(article_obj: &Object<Article>): String {
+        object::borrow(article_obj).title
+    }
+
+    public(friend) fun set_title(article_obj: &mut Object<Article>, title: String) {
+        object::borrow_mut(article_obj).title = title;
+    }
+
+    public fun body(article_obj: &Object<Article>): String {
+        object::borrow(article_obj).body
+    }
+
+    public(friend) fun set_body(article_obj: &mut Object<Article>, body: String) {
+        object::borrow_mut(article_obj).body = body;
+    }
+
+    fun new_article(
+        _tx_ctx: &mut tx_context::TxContext,
+        title: String,
+        body: String,
+    ): Article {
+        assert!(std::string::length(&title) <= 200, EID_DATA_TOO_LONG);
+        assert!(std::string::length(&body) <= 2000, EID_DATA_TOO_LONG);
+        Article {
+            version: 0,
+            title,
+            body,
+        }
+    }
+
+    struct ArticleCreated has key {
```

Comment:
This is an event. Rename it to `ArticleCreatedEvent` and remove the `key` ability.